### ISSUE TITLE
specification for values in Racing Lab screen

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/unit-rates#143]
Specification for the values (numerator and denominator) in the Racing Lab screen is not totally clear in the design document.

Here's my best guess, based on doc, screenshots and old implementation:

Numerator (miles)
• maximum 4 digits and 2 decimal places  (e.g. 1234, 123.4, 12.34)
• strip off zeros in decimal places? Yes (e.g. 55.20 is displayed as 55.2; 55.0 is displayed as 55)

Denominator (hours)
• maximum 4 digits and 2 decimal places  (e.g. 1234, 123.4, 12.34)
• strip off zeros in decimal places? No (eg. 10.2 is displayed as 10.20, 10 is displayed as 10.00)

And I'm assuming that the same specification applies everywhere. I.e., the timer and the bottom axis of the double number line display hours in the same format.

@arouinfar Please verify or correct.

**Comment by pixelzoom** [comment]
The above specification is implemented in this dev version:
http://www.colorado.edu/physics/phet/dev/html/unit-rates/1.0.0-dev.42/unit-rates_en.html?showAnswers

Go to the new Racing Lab screen, and create markers using the marker editor. (Ignore other aspects of this screen, it's in a very early stage of implementation.)

A problem that I noticed already, demonstrated by example:
1. Go to Racing Lab screen
2. Use the marker editor to enter denominator 5555.
3. The corresponding numerator (555500) is impossible to enter, because it requires more than 4 digits.

**Comment by arouinfar** [comment]
My initial recommendation would be what you outlined in https://github.com/phetsims/unit-rates/issues/143#issue-205703801 @pixelzoom. However, you've found an excellent counter-example. :)

Perhaps instead we should limit the number of hours to a maximum of 99.99 (so 9999 and 999.9 would no longer be allowed). That way, regardless of the rate, the corresponding numerator could always be entered. For example, at 100 mph, the corresponding distance would be 9999 miles.

However, there will be syntactically-allowed numerators that will correspond to impossible-to-enter denominators. For example, at 2 mph, a numerator of 1000 miles, corresponds to a denominator of 500 miles, which would no longer be allowed. There are already situations like this on the Shopping screen. For example, $9999 corresponds to an impossible-to-enter amount of purple candy ($1851.67). 

To summarize, I suppose the only change I would make to @pixelzoom's original suggestion would be to limit the number of hours to a maximum of 2 digits before the decimal point and 2 digits after the decimal point (eg. 1.2, 1.23, 12.3, 12.34).

**Comment by pixelzoom** [comment]
@arouinfar said:
> To summarize, I suppose the only change I would make to @pixelzoom's original suggestion would be to limit the number of hours to a maximum of 2 digits before the decimal point and 2 digits after the decimal point (eg. 1.2, 1.23, 12.3, 12.34).

Limiting the number of the digits _before_ the decimal point is not supported by the common-code keypad (scenery-phet.NumberKeypad) -- the only constraints currently supported are total number of digits and number of decimal places.  A new keypad is under development (see https://github.com/phetsims/scenery-phet/issues/283), but it's not expected to be completed before Unit Rates is published.  So if you want this behavior, I'll need to write a new "validator" for the existing common Keypad.

@arouinfar Let me know how you'd like to proceed with this.

**Comment by pixelzoom** [comment]
2/9/17 design meeting:
• strip off zeros for denominator (hours)

**Comment by pixelzoom** [comment]
This specification is reflected in
http://www.colorado.edu/physics/phet/dev/html/unit-rates/1.0.0-dev.45/unit-rates_en.html

@arouinfar please review and close if this looks OK.

**Comment by arouinfar** [comment]
Looks great!
